pkg/attest/platform_cert_fetcher.go

```diff
@@ -166,33 +173,36 @@ func fetchWithRetry(requestURL string, baseSec int, maxRetries int) ([]byte, err
 		}
 		res, err := http.Get(requestURL)
 		if err != nil {
+			logrus.Debugf("fetch on retry %d: http.Get failed: %s", retryCount, err)
 			retryCount++
 			continue
 		}
 		if 200 <= res.StatusCode && res.StatusCode < 300 {
 			// Got successful status code 2xx
 			defer res.Body.Close()
-			resBody, err := ioutil.ReadAll(res.Body)
+			resBody, err := io.ReadAll(res.Body)

```

Comment:
Be careful with `io.ReadAll`.  A malicious proxy can overflow your memory by responding with unlimited bytes.  In general you should pass the reader back to the caller to deal with the response directly, or wrap it in a LimitReader: https://pkg.go.dev/io#LimitReader.